hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/unguava/Validate.java

```diff
@@ -0,0 +1,123 @@
+ *   <li>An invalid state causes an {@link IllegalStateException}.</li>
+ *   <li>An invalid index causes an {@link IndexOutOfBoundsException}.</li>
+ * </ul>
+ */
+@InterfaceAudience.Private
+@InterfaceStability.Unstable
+public final class Validate {
+  public static final Logger LOG =
+      LoggerFactory.getLogger(Validate.class);
+
+  public static final String VALIDATE_IS_NOT_NULL_EX_MESSAGE =
+      "The validated object is null";
+
+  private Validate() {
+  }
+
+  /**
+   * <p>Validate that the specified argument is not {@code null},
+   * throwing a NPE exception otherwise.
+   *
+   * <p>The message of the exception is
+   * &quot;The validated object is null&quot;.</p>
+   *
+   * @param <T> the object type
+   * @param obj  the object to check
+   * @return the validated object
+   * @throws NullPointerException if the object is {@code null}
+   * @see #checkNotNull(Object, String, Object...)
+   */
+  public static <T> T checkNotNull(final T obj) {
+    return checkNotNull(obj, VALIDATE_IS_NOT_NULL_EX_MESSAGE);
+  }
+
+  public static <T> T checkNotNull(final T reference,
+      final Object errorMessage) {
+    return Objects.requireNonNull(reference, String.valueOf(errorMessage));
+  }
+
+  /**
+   * <p>Validate that the specified argument is not {@code null},
+   * throwing a NPE exception otherwise.
+   *
+   * <p>The message of the exception is
+   * &quot;The validated object is null&quot;.</p>
+   *
+   * @param <T> the object type
+   * @param obj  the object to check
+   * @param message  the {@link String#format(String, Object...)}
+   *                 exception message if valid. Otherwise,
+   *                 the message is {@link #VALIDATE_IS_NOT_NULL_EX_MESSAGE}
+   * @param values  the optional values for the formatted exception message
+   * @return the validated object
+   * @throws NullPointerException if the object is {@code null}
+   * @see #checkNotNull(Object, Supplier<String>)
+   */
+  public static <T> T checkNotNull(final T obj, final String message,
+      final Object... values) {
+    return checkNotNull(obj, () -> String.format(message, values));
+  }
+
+  /**
+   * <p>Validate that the specified argument is not {@code null},
+   * throwing a NPE exception otherwise.
+   *
+   * <p>The message of the exception is
+   * &quot;The validated object is null&quot;.</p>
+   *
+   * @param <T> the object type
+   * @param obj  the object to check
+   * @param msgSupplier  the {@link Supplier#get()} set the
+   *                 exception message if valid. Otherwise,
+   *                 the message is {@link #VALIDATE_IS_NOT_NULL_EX_MESSAGE}
+   * @return the validated object (never {@code null} for method chaining)
+   * @throws NullPointerException if the object is {@code null}
+   * @see #checkNotNull(Object, Supplier<String>)
+   */
+  public static <T> T checkNotNull(final T obj,
+      final Supplier<String> msgSupplier) {
+    if (obj == null) {
```

Comment:
There's an `Objects.requireNonNull(T obj, Supplier<String> messageSupplier)` which you could forward to, but your implementation is actually more robust, so I think it should stay

Reply:
Yes, I thought that guava.checkNonNull swallows ill-formatted messages by calling `lenientFormat` internally. So, I wanted to handle those cases to imitate the same behavior.